apps/site/components/withDownloadArchive.tsx

```diff
@@ -0,0 +1,45 @@
+import { notFound } from 'next/navigation';
+import type { FC } from 'react';
+import semVer from 'semver';
+
+import { getClientContext } from '#site/client-context';
+import provideReleaseData from '#site/next-data/providers/releaseData';
+import {
+  buildReleaseArtifacts,
+  extractVersionFromPath,
+} from '#site/util/download/archive';
+
+type DownloadArchive = ReturnType<typeof buildReleaseArtifacts>;
+
+type WithDownloadArchiveProps = {
+  children: FC<DownloadArchive>;
+};
+
+/**
+ * Higher-order component that extracts version from pathname,
+ * fetches release data, and provides download artifacts to child component
+ */
+const WithDownloadArchive: FC<WithDownloadArchiveProps> = async ({
+  children: Component,
+}) => {
+  const { pathname } = getClientContext();
+
+  // Extract version from pathname
+  const version = extractVersionFromPath(pathname);
+
+  if (version == null) {
```

Comment:
Will a situation where users can reach on this Provider with this being null ever happen? I just worry that this isn't really needed. I think you can remove this null statement fully. Otherwise, let me know in which situation this happened 🤔